.ci/test-r-package.sh

```diff
@@ -19,13 +19,13 @@ if [[ $R_BUILD_TYPE != "cran" ]]; then
 fi
 
 # Get details needed for installing R components
-R_MAJOR_VERSION=( ${R_VERSION//./ } )
-if [[ "${R_MAJOR_VERSION}" == "3" ]]; then
+IFS='.' read -ra R_MAJOR_VERSION <<< "$R_VERSION"
```

Comment:
This doesn't look right to me. I'm very uncomfortable with modifying `IFS` in scope for the rest of this script... that can affect all commands that follow.

What is the specific `shellcheck` warning this is intended to address?

Reply:
The message is SC2206 (warning): Quote to prevent word splitting/globbing, or split robustly with mapfile or read -a.

SC2206 (warning): Quote to prevent word splitting/globbing, or split robustly with mapfile or read -a.

Reply:
Ok thanks! Instead of that, let's avoid reading this into an array and just hard-code the assumption that `R_VERSION` will be a 3-part semantic version. It has been for 20+ years (https://cran.r-project.org/bin/windows/base/old/).

So that would mean doing this

```bash
R_MAJOR_VERSION="${R_VERSION%.*.*}"
```

Per https://www.gnu.org/software/bash/manual/bash.html

> `${parameter%word}`
> `${parameter%%word}`
> *The word is expanded to produce a pattern and matched according to the rules described below (see [Pattern Matching](https://www.gnu.org/software/bash/manual/bash.html#Pattern-Matching)). If the pattern matches a trailing portion of the expanded value of parameter, then the result of the expansion is the value of `parameter` with the shortest matching pattern (the `%` case) or the longest matching pattern (the `%%` case) deleted.*

This is totally safe... if the assumption is ever violated (for example we test against an R `4.6.8.1` or something), this block will alert us via a loud error:

https://github.com/microsoft/LightGBM/blob/41ba9e8f00c89d72e5cb71c964722ce1ed4d8445/.ci/test-r-package.sh#L33-L36

Reply:
Noted. Thanks. Adressed. Indeed if we can work with assumptions on the R string version format, let's simplify the parsing!